src/transports/tcp/TcpDuplexConnection.cpp

```diff
@@ -64,20 +76,20 @@ class TcpReaderWriter : public ::folly::AsyncTransportWrapper::WriteCallback,
     if (auto socket = std::move(socket_)) {
       socket->close();
     }
+    if (auto outputSubscription = std::move(outputSubscription_)) {
```

Comment:
Why is `subscription_->cancel()` moved up above `subscriber_->onComplete()` here in `close(), but not in `closeErr()`?